indico/MaKaC/PDFinterface/conference.py

```diff
@@ -1439,112 +1440,65 @@ def firstPage(self, c, doc):
-                    text += " ; ".join(listSocialEvents)
-            elif key == "Accommodation" and self._reg.getAccommodation() is not None and \
-            self._reg.getAccommodation().getAccommodationType() is not None:
-                text = i18nformat("""<b> _("Acommodation")</b>: %s""") % escape(self._reg.getAccommodation().getAccommodationType().getCaption())
-            elif key == "ArrivalDate" and self._reg.getAccommodation() is not None and \
-            self._reg.getAccommodation().getArrivalDate() is not None:
-                text = i18nformat("""<b> _("Arrival Date")</b>: %s""") % escape(self._reg.getAccommodation().getArrivalDate().strftime("%d-%B-%Y"))
-            elif key == "DepartureDate" and self._reg.getAccommodation() is not None and \
-            self._reg.getAccommodation().getDepartureDate() is not None:
-                text = i18nformat("""<b> _("Departure Date")</b>: %s""") % escape(self._reg.getAccommodation().getDepartureDate().strftime("%d-%B-%Y"))
-            elif key == "ReasonParticipation" and self._reg.getReasonParticipation() is not None and \
-            self._reg.getReasonParticipation() is not None and self._reg.getReasonParticipation() != "":
-                text = i18nformat("""<b> _("Reason Participation")</b>: %s""") % escape(self._reg.getReasonParticipation())
-            elif key == "RegistrationDate" and self._reg.getRegistrationDate() is not None:
-                text = i18nformat("""<b> _("Registration date")</b>: %s""") % escape(self._reg.getAdjustedRegistrationDate().strftime("%d-%B-%Y-%H:%M"))
-
-            elif key.startswith("s-"):
-                ids = key.split("-")
-                if len(ids) == 2:
-                    status = self._reg.getStatusById(ids[1])
-                    if status.getStatusValue() is not None:
-                        text = _("""<b> %s</b>: %s""") % (escape(status.getCaption()), escape(status.getStatusValue().getCaption()))
-            else:
-                ids = key.split("-")
-                if len(ids) == 2:
-                    group = self._reg.getMiscellaneousGroupById(ids[0])
-                    if group is not None:
-                        field = self._conf.getRegistrationForm().getSectionById(ids[0]).getFieldById(ids[1])
-                        response = group.getResponseItemById(ids[1])
-                        if response is not None and response.getValue() != "":
-                            text = _("""<b> %s</b>: %s""") % (escape(field.getCaption()), escape(str(response.getValue())))
-            if text != "":
-                p = Paragraph(text, style,  part=escape(self._reg.getFullName()))
-                story.append(p)
-                story.append(Spacer(inch, 0.2*cm, part=escape(self._reg.getFullName())))
+
+        for item in self._display:
+            value = data[item.id].friendly_data if item.id in data else ''
+            if isinstance(value, basestring):
```

Comment:
`unicode` instead of `basestring` - if it's already a str you can assume it's utf8 (if it's not trying to encode it as utf8 would fail anyway since it'd implicitly try to decode it as ascii first)